pkg/tfbridge/provider.go

```diff
@@ -770,6 +782,36 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 		return nil, err
 	}
 
+	logger := GetLogger(ctx)
+	// for now we are just going to log warnings if there are failures.
+	// over time we may want to turn these into actual errors
+	_, validateShouldError := os.LookupEnv("PULUMI_ERROR_TYPE_CHECKER")
```

Comment:
`PULUMI_ERROR_TYPE_CHECKER=false` should not be equivalent to `PULUMI_ERROR_TYPE_CHECKER=true`.
```suggestion
	validateShouldError := "github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil".IsTruthy(os.Getenv("PULUMI_ERROR_TYPE_CHECKER"))
```

Reply:
Updated!